src/webserver/routes/api/store.ts

```diff
@@ -1429,23 +1438,22 @@ router.post(
 					manager: req.user.id,
 				},
 				reason: `${item.name}${quantity > 1 ? ` x${quantity}` : ''}`,
-			}).save();
-
+			};
+			let t = await new Transaction(transactionData).save();
 			req.user.balance -= price;
 			await req.user.save();
 
-			let request = await new StoreRequest({
+			let requestData = await new StoreRequest({
 				storeID: store._id,
 				itemID: item._id,
 				studentID: req.user._id,
-				transactionID: transaction._id,
+				transactionID: t._id,
 				quantity,
 				price,
 			}).save();
-
 			await item.save();
-
-			res.status(200).json(await request.toAPIResponse());
+			queue.add('request', transactionData);
```

Comment:
I don't like that you're sending `transactionData` to the worker, and I think you should adapt the code to use `requestData` instead.

The use is problematic for 2 reasons:

1. The transaction here is just an ordinary transaction, while the request document exists for this exact purpose.
2. This part isn't your fault, but if you look at where we define `transactionData`, you'll notice that the `to` and the `store.manager` properties are both set to the user that initiated the request (the student). We're claiming that the student is also the manager! This is not right! While you could just fix this, I still take issue with the fact that such a property only accepts one user, while Kitcoin supports multiple store managers.

What this ends up doing is emailing the person who made the request instead of the student:
![image](https://github.com/codeths/kitcoin/assets/37093293/b7442bd5-c5b3-4a28-878c-277e8e08198f)

I think that you should use the request data instead of the transaction data, and when you go to send the email, determine who the managers are for the store and email all of them.

Reply:
From what I can tell, this doesn't provide a list of all users that can manage the store, does it?

Reply:
> From what I can tell, this doesn't provide a list of all users that can manage the store, does it?

My bad. You could probably use the store owner and managers from the db for now, and we can worry about emailing classroom teachers later. Not ideal, but at least every store has a owner to email...